engine/swarm/secrets.md

```diff
@@ -0,0 +1,770 @@
+with access to a secret, the task container still has access to its secrets, but
+cannot receive updates until the node reconnects to the swarm.
+
+You can add or inspect an individual secret at any time, or list all
+secrets. You cannot remove a secret that a running service is
+using. See [Rotate a secret](secrets.md#example-rotate-a-secret) for a way to
+remove a secret without disrupting running services.
+
+In order to update or roll back secrets more easily, consider adding a version
+number or date to the secret name. This is made easier by the ability to control
+the mount point of the secret within a given container.
+
+## Read more about `docker secret` commands
+
+Use these links to read about specific commands, or continue to the
+[example about using secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+- [`docker secret create`](../reference/commandline/secret_create.md)
+- [`docker secret inspect`](../reference/commandline/secret_inspect.md)
+- [`docker service ls`](../reference/commandline/secret_ls.md)
+- [`docker secret rm`](../reference/commandline/secret_rm.md)
+- [`--secret`](../reference/commandline/service_create.md#create-a-service-with-secrets) flag for `docker service create`
+- [`--secret-add` and `--secret-rm`](../reference/commandline/service_update.md#adding-and-removing-secrets) flags for `docker service update`
+
+## Examples
+
+This section includes three graduated examples which illustrate how to use
+Docker secrets. The images used in these examples have been updated to make it
+easier to use Docker secrets. To find out how to modify your own images in
+a similar way, see
+[Build support for Docker Secrets into your images](#build-support-for-docker-secrets-into-your-images).
+
+### Simple example: Get started with secrets
+
+This simple example shows how secrets work in just a few commands. For a
+real-world example, continue to
+[Use secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+1.  Add a secret to Docker. The `docker secret create` command reads standard
+    input.
+
+    ```bash
+    $ echo "This is a secret" | docker secret create my_secret_data -f -
+    ```
+
+2.  Create a `redis` service and grant it access to the secret. By default,
+    the container can access the secret at `/run/secrets/<secret_name>`, but
+    you can customize the file name on the container using the `target` option.
+
+    ```bash
+    $ docker service  create --name="redis" --secret="my_secret_data" redis:alpine
+    ```
+
+3.  Verify that the task is running without issues using `docker service ps`. If
+    everything is working, the output looks similar to this:
+
+    ```bash
+    $ docker service ps --no-trunc redis
+
+    NAME                               lIMAGE         NODE              DESIRED STATE  CURRENT STATE           ERROR  PORTS
+    redis.1.cezsyv6xljjj6bzpajbc6agkb  redis:alpine   ip-172-31-46-109  Running        Running 26 seconds ago
```

Comment:
The output format of `docker service ps` changed back to the 1.12 format, so no `--no-trunc` is needed;

```bash
$ docker service ps web

ID            NAME   IMAGE         NODE          DESIRED STATE  CURRENT STATE          ERROR  PORTS
kht7jvzvai5z  web.1  nginx:alpine  371b354ecd17  Running        Running 3 seconds ago
```